### PR TITLE
Add `extract_*()` generics

### PR DESCRIPTION
**Pull request by DavisVaughan**
- [x] expand on what an "engine fit" is

### CODE REVIEW

**Review by topepo — COMMENTED** [review]
Can you do one man file per function and then use calls like

```r
#' @section Methods:
#' \Sexpr[stage=render,results=rd]{generics:::methods_rd("extract_workflows")}
```

**Review by topepo — COMMENTED** [review]
I rescind my suggestion for the `generics`-type documentation.

File: R/extract.R [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#' objects. Methods are defined in other packages, such as tune,
+#' workflows, and workflowsets, but the returned object is always the same.
+#'
+#' - `extract_workflow()` returns a workflow, possibly fit.
+#'
+#' - `extract_recipe()` returns a recipe, possibly prepped.
+#'
+#' - `extract_parsnip_spec()` returns a parsnip model specification.
+#'
+#' - `extract_parsnip_fit()` returns a parsnip model fit.
```
[the remark this answers is not in the document]
Comment (topepo) [reply R/extract.R]: Resolved in c487a38

File: R/extract.R [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+#' objects. Methods are defined in other packages, such as tune,
+#' workflows, and workflowsets, but the returned object is always the same.
+#'
+#' - `extract_fit_engine()` returns the engine specific fit embedded within
+#'   a parsnip model fit. For example, when using [parsnip::linear_reg()]
+#'   with the `"lm"` engine, this would return the underlying `lm` object.
+#'
+#' - `extract_fit_parsnip()` returns a parsnip model fit.
+#'
+#' - `extract_mold()` returns returns the preprocessed "mold" object returned
```
Comment (DavisVaughan) [inline R/extract.R:14]:
```suggestion
#' - `extract_mold()` returns the preprocessed "mold" object returned
```

Comment (DavisVaughan) [reply R/extract.R:14]: plus a redocument locally

**Pull request merged**